### PR TITLE
Feature/filtering by namespace and datasource

### PR DESCRIPTION
**Pull request by mell0kat**
### Description

This PR implements filtering by namespace and datasource.

Includes:
* `<FilterContainer/>`
* `<Filters/>` component <-- majority of changes
* Modifies jobs & datasets requests to return namespace on the resource
* adds `FILTER_JOBS` and `FILTER_DATASETS` actions and actionCreators, as well as associated reducers and tests
* `filterEntities` helper function for the above ^ 
* Updates docker/db/data/ to use more than one data source

filtering

### Checklist

- [x] This PR contains tests (at least one (non-snapshot) test)

### CODE REVIEW

File: src/__tests__/reducers/index.test.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { findMatchingEntities } from '../../reducers'
-import { match } from 'minimatch'
+import { findMatchingEntities, filterEntities } from '../../reducers'
+
 describe('findMatchingEntities test', () => {
   const datasets = require('../../../docker/db/data/datasets.json')
   const matchingDatasets = findMatchingEntities('searchTerm', datasets)
```
Comment (grantdfoster) [inline src/__tests__/reducers/index.test.ts]: i  guess we should write this test...

File: src/components/Filters.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      accessor: (n: string): string => n,
+      default: ''
+    }
+    /* Can add more filters here */
+  }
+
+  /* Set the category we will be filtering by */
+  const onPrimaryFilterChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    const newPrimaryFilter = e.target.value as 'namespace' | 'datasource' | 'all'
+    console.log('NEW PRI<AR', newPrimaryFilter)
```
Comment (grantfosterdev) [inline src/components/Filters.tsx]: found a console log fyi!

File: src/components/Filters.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    setCurrentFilter(newPrimaryFilter)
+    if (newPrimaryFilter === 'all') {
+      toggleShowSubFilter(false)
```
Comment (grantfosterdev) [inline src/components/Filters.tsx]: maybe rename this for clarity?  I would assume a toggle function wouldn't take an argument?

File: src/components/Filters.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+const filterByOptions: { [key: string]: 'namespace' | 'sourceName' } = {
+  /* display name: entitiy name on dataset schema */
+  namespace: 'namespace',
+  datasource: 'sourceName'
+}
+
+const Filters = (props: IProps): ReactElement => {
+  const { namespaces, datasets, filterJobs, filterDatasets, showJobs } = props
+
+  const [currentFilter, setCurrentFilter] = useState('all')
```
Comment (grantfosterdev) [inline src/components/Filters.tsx:38]: definitely wouldn't mind a crash course on React hooks (I assume this is what this is?)

File: src/components/Filters.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  const filterDictionary: IFilterDictionary = {
+    namespace: {
+      entities: namespaces,
+      accessor: n => (n as INamespaceAPI).name,
+      default: { name: '' }
+    },
+    datasource: {
+      entities: datasources,
+      accessor: (n: string): string => n,
+      default: ''
```
Comment (grantfosterdev) [inline src/components/Filters.tsx:52]: why don't we need the `.name` attribute here?  Aren't namespaces structured the same as our datasets?

**Review by geeeeemoney — COMMENTED** [review]
I left a few comments (more just conversational based) but looks great!  Lots of clean organization

File: src/components/Filters.tsx (hunk 3 above)
Comment (mell0kat) [reply src/components/Filters.tsx]: Totally agree with this!  Will change.

File: src/components/Filters.tsx (hunk 2 above)
Comment (mell0kat) [reply src/components/Filters.tsx]: Good catch!

File: src/components/Filters.tsx (hunk 4 above)
Comment (mell0kat) [reply src/components/Filters.tsx:38]: yes exactly!  Honestly, the docs are pretty great -- https://reactjs.org/docs/hooks-intro.html

File: src/components/Filters.tsx (hunk 5 above)
Comment (mell0kat) [reply src/components/Filters.tsx:52]: so this is actually datasource, which is just `sourceName`, which is a string on the dataset field.  Does this make sense?

**Review by geeeeemoney — APPROVED** [review]

**Comment by mell0kat** [comment]
@jhubley Fixed the margin issue
Screen Shot 2019-11-20 at 11 20 15 AM

**Review by jhubley — APPROVED** [review]
Nice

**Pull request merged**